### PR TITLE
make use of Streams instead of VersionIterator

### PR DESCRIPTION
**Pull request by acoburn**
As @ajs6f suggested, this makes use of a more Stream-based approach.

### CODE REVIEW

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java [hunk 1]
```diff
@@ -823,24 +822,17 @@ public String getVersionLabelOfFrozenResource() {
...
-                    if (labels.length > 0) {
-                        return labels[0];
-                    }
-                }
-            }
+            @SuppressWarnings("unchecked")
+            final Stream<Version> versions = iteratorToStream(versionHistory.getAllVersions());
+            return versions
+                .filter(UncheckedPredicate.uncheck(version -> version.getFrozenNode().equals(frozenResource)))
+                .map(uncheck(version -> versionHistory.getVersionLabels(version)))
```
Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: `uncheck(versionHistory::getVersionLabels)`?

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java [hunk 2]
```diff
@@ -823,24 +822,17 @@ public String getVersionLabelOfFrozenResource() {
...
+                .filter(labels -> labels.length > 0)
```
Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: `ArrayUtils::isEmpty`?

**Comment by ajs6f** [comment]
I was actually okay with an iterator and `forEachRemaining`, but this does read pretty well.

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java (hunk 2 above)
Comment (acoburn) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: we actually want the inverse of `isEmpty`.

Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: Ach-- of course, my bad! So, `ArrayUtils::isNotEmpty`.

**Comment by awoods** [comment]
I had a very similar implementation originally, but with the three flavors of `unchecked` cluttering the code, the Java7 rendition was more pleasing to my eye.
...but your effort does not go unappreciated. Thanks.

**Pull request merged**